tests/instancetype/upgrade.go

```diff
@@ -75,29 +76,23 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 	}
 
 	updateInstancetypeMatcher := func(revisionName string) {
-		Eventually(func(g Gomega) {
-			var err error
-			vm, err = virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Get(context.Background(), vm.Name, metav1.GetOptions{})
-			g.Expect(err).ToNot(HaveOccurred())
-
-			vm.Spec.Instancetype.RevisionName = revisionName
+		vmPatch, err := patch.New(
+			patch.WithReplace("/spec/instancetype/revisionName", revisionName),
+		).GeneratePayload()
+		Expect(err).ToNot(HaveOccurred())
 
-			_, err = virtClient.VirtualMachine(vm.Namespace).Update(context.Background(), vm, metav1.UpdateOptions{})
-			g.Expect(err).ToNot(HaveOccurred())
-		}, 30*time.Second, time.Second).Should(Succeed())
+		_, err = virtClient.VirtualMachine(vm.Namespace).Patch(context.Background(), vm.Name, types.JSONPatchType, vmPatch, metav1.PatchOptions{})
```

Comment:
I didn't look at it too deeply but for consistency with the previous behavior, you should set the `vm` variable with the returned value, so that the `/spec/instancetype/revisionName` is properly valued.

Reply:
I wasn't aware that the vm is a global variable here. Fixed.